alicloud/resource_alicloud_cs_kubernetes.go

```diff
@@ -778,21 +832,35 @@ func buildKubernetesArgs(d *schema.ResourceData, meta interface{}) (*cs.Kubernet
 		LoginPassword:            d.Get("password").(string),
 		KeyPair:                  d.Get("key_name").(string),
 		Network:                  d.Get("cluster_network_type").(string),
+		NodeCIDRMask:             d.Get("node_cidr_mask").(string),
+		LoggingType:              loggingType,
+		SLSProjectName:           slsProjectName,
 		NumOfNodes:               int64(workerNumber),
 		MasterSystemDiskCategory: ecs.DiskCategory(d.Get("master_disk_category").(string)),
 		MasterSystemDiskSize:     int64(d.Get("master_disk_size").(int)),
 		WorkerSystemDiskCategory: ecs.DiskCategory(d.Get("worker_disk_category").(string)),
 		WorkerSystemDiskSize:     int64(d.Get("worker_disk_size").(int)),
-		WorkerDataDiskCategory:   d.Get("worker_data_disk_category").(string),
-		WorkerDataDiskSize:       int64(d.Get("worker_data_disk_size").(int)),
 		SNatEntry:                d.Get("new_nat_gateway").(bool),
 		KubernetesVersion:        d.Get("version").(string),
 		ContainerCIDR:            d.Get("pod_cidr").(string),
 		ServiceCIDR:              d.Get("service_cidr").(string),
 		SSHFlags:                 d.Get("enable_ssh").(bool),
 		CloudMonitorFlags:        d.Get("install_cloud_monitor").(bool),
 		ZoneId:                   zoneId,
-	}, nil
+	}
+
+	if v, ok := d.GetOk("worker_data_disk_category"); ok {
+		creationArgs.WorkerDataDiskCategory = v.(string)
+		creationArgs.WorkerDataDisk = true
+		if size, ok := d.GetOk("worker_data_disk_size"); ok {
+			creationArgs.WorkerDataDiskSize = int64(size.(int))
+		} else {
+			creationArgs.WorkerDataDiskSize = 40
+		}
```

Comment:
I think both `worker_data_disk_category` and `worker_data_disk_size` must be set at the same time, or do not.

Reply:
worker_data_disk_size are 40 by default when worker_data_disk_category is set.